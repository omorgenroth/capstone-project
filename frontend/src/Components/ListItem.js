import styled from 'styled-components/macro'
import { FaMinus } from 'react-icons/fa'

export default function ListItem({ title }) {
  return (
    <ItemWrapper>
      <IconWrapper>
        <Circle>
          <DeleteIcon />
        </Circle>
      </IconWrapper>
      <ContentWrapper>
        <Title>{title}</Title>
      </ContentWrapper>

      <Ruler />
    </ItemWrapper>
  )
}

const ItemWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 4fr;
  grid-template-rows: 3fr 1fr;
`
const Title = styled.h2`
  font-size: 1rem;
`

const ContentWrapper = styled.div`
  grid-column: 2/3;
`

const IconWrapper = styled.div`
  grid-column: 1/2;
`
const DeleteIcon = styled(FaMinus)`
  color: var(--c-gray);
  transform: scale(1.5);
  padding-right: 10px;
  z-index: 5;
`

const Ruler = styled.hr`
  border: 0.3px solid var(--c-green);
  grid-column: 1/-1;
  grid-row: 2/3;
`
const Circle = styled.div`
  border-radius: 50%;
  border: none;
  background: #ff6e4a;
  width: 20px;
`
