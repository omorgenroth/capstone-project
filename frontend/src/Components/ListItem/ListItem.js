import { FaMinusCircle } from 'react-icons/fa'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

ListItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  onDelete: PropTypes.func,
}

export default function ListItem({ id, title, onDelete }) {
  return (
    <>
      <ItemWrapper>
        <DeleteIcon onClick={() => onDelete(id)} />
        <Title>{title}</Title>
      </ItemWrapper>
      <Ruler />
    </>
  )
}

const ItemWrapper = styled.div`
  display: flex;
`

const Title = styled.div`
  font-size: 1rem;
`

const DeleteIcon = styled(FaMinusCircle)`
  color: var(--c-orange);
  transform: scale(1);
  margin-right: 5px;
`

const Ruler = styled.hr`
  border: 1px solid var(--c-green);
  width: 85%;
`
