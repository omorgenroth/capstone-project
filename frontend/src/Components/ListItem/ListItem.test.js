import { render } from '@testing-library/react'
import ListItem from './ListItem'

describe('ListItem', () => {
  it('renders correctly', () => {
    const { container } = render(<ListItem title="My Dish" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
