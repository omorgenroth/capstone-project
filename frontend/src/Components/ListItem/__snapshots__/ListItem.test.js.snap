// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  font-size: 1rem;
}

.c1 {
  color: var(--c-orange);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  margin-right: 5px;
}

<div
  class="c0"
>
  <svg
    class="c1"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z"
    />
  </svg>
  <div
    class="c2"
  >
    My Dish
  </div>
</div>
`;
